### DISCUSSION TITLE
posts/2024/03/thanks-goodbye-cloudinary/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#45] [Announcements]
# posts/2024/03/thanks-goodbye-cloudinary/

After my code exceeds the limits of an already overly generous Free Plan, I again process all images locally.

https://www.brycewray.com/posts/2024/03/thanks-goodbye-cloudinary/?utm_campaign=RSS&utm_source=RSS&utm_medium=RSS

**Comment by idarek** [comment]
I am really glad that Hugo got an option to generate images locally. I am pushing `respources/_gen/images` with my Git repo, so they are not re-generated when built (in my instance with Netlify) and it simply works. Build times are short and all is exactly as on the local device. Off course, pushing this files to Github may be strange for somehow, but at least I got full control on that.

**Reply by brycewray** [reply]
@idarek Yes, it’s wonderful to have the built-in capability, especially when — as I described in this post — it becomes necessary to go it alone. Back when I [first started using Cloudinary](https://www.brycewray.com/posts/2020/07/transformed/), I not only was using Eleventy **before** the existence of the Eleventy Image plugin but also relied on [my own working-but-slow **build-time** JavaScript](https://github.com/brycewray/eleventy-site/blob/3752522144af9f3c138306e5b33aa831b93a0479/imgxfm.js) for image processing, so just about anything would’ve been better than that. But, having taken time since then to learn how Hugo can do it on its own, I’ve usually been quite happy to opt for that. I **definitely** am now.
